tools/reviewer/web/src/app/dashboard/reviews-panel/review/person-list/person-list.component.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  Diff,
+  FirebaseService,
+  NotificationService,
+  ProtoService,
+  Status
+} from '@/shared';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-person-list',
+  templateUrl: './person-list.component.html',
+  styleUrls: ['./person-list.component.scss']
+})
+export class PersonListComponent {
+  @Input() diff: Diff;
+  @Input() property: string;
+  @Input() addToAttention: boolean = true;
+  @Output() onUpdateDiff = new EventEmitter<Diff>();
+
+  // Fields can not be edited if status is
+  // 'SUBMITTED' or 'REVERTED'
+  editable: boolean = true;
+  notEditableStatus: Array<number> = [Status.SUBMITTED, Status.REVERTED];
+
+  // Following variables are used to show editable fields
+  showEditableProperty = false;
+
+  // Following variables are used in editing
+  // the fields
+  reviewers: string = '';
+  cc: string = '';
+
+  constructor(
+    private protoService: ProtoService,
+    private firebaseService: FirebaseService,
+    private notify: NotificationService
+  ) {}
+
+  ngOnChanges() {
+    this.getPropertyValue(this.property);
+    // Render the fields un-editable if the current diff status
+    // is in the list of notEditableStatus
+    this.editable = !this.notEditableStatus.includes(this.diff.status);
+  }
+
+  // Get property value from the Diff
+  getPropertyValue(property: string): void {
+    this[property] = this.diff[property].join(', ');
```

Comment:
What does `this[property]` mean? Is `this` a map?

Reply:
`this[property]` enables us to use `this['reviewers'] and `this['cc']`  based on the value of property. Hope this also explains the use of `property`